src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1459 @@
+            institution_name=institution_name,
+            institutional_department_name=institutional_department_name,
+            content_creator_name=content_creator_name,
+            content_creator_identification=content_creator_identification,
+            rescale_intercept=rescale_intercept,
+            rescale_slope=rescale_slope,
+            rescale_type=rescale_type,
+            modality_lut=modality_lut,
+            copy_modality_lut=copy_modality_lut,
+            softcopy_voi_luts=softcopy_voi_luts,
+            copy_voi_lut=copy_voi_lut,
+            icc_profile=icc_profile,
+            transfer_syntax_uid=transfer_syntax_uid,
+            **kwargs
+        )
+
+        # Palette Color Lookup Table
+        self._add_palette_color_lookup_table(palette_color_lut)
+
+    def _add_palette_color_lookup_table(
+        self,
+        palette_color_lut: PaletteColorLookupTable
+    ) -> None:
+        """Add attributes from the Palette Color Lookup Table module.
+
+        Parameters
+        ----------
+        palette_color_lut: highdicom.content.PaletteColorLookupTable
+            A palette color lookup table to apply to the image.
+
+        """
+        if not isinstance(palette_color_lut, PaletteColorLookupTable):
+            raise TypeError(
+                'Argument "palette_color_lut" must be of type '
+                'PaletteColorLookupTable.'
+            )
+        colors = ['Red', 'Green', 'Blue']
+
+        for color in colors:
+            desc_kw = f'{color}PaletteColorLookupTableDescriptor'
+            data_kw = f'{color}PaletteColorLookupTableData'
+            desc = getattr(palette_color_lut, desc_kw)
+            lut = getattr(palette_color_lut, data_kw)
+
+            setattr(self, desc_kw, desc)
+            setattr(self, data_kw, lut)
+
+        if hasattr(palette_color_lut, 'PaletteColorLookupTableUID'):
+            uid = palette_color_lut.PaletteColorLookupTableUID
+            self.PaletteColorLookupTableUID = uid
+
+
+class ColorSoftcopyPresentationState(_SoftcopyPresentationState):
+
+    """SOP class for a Color Softcopy Presentation State object.
+
+    A Color Softcopy Presentation State object includes instructions for
+    the presentation of a color image by software.
+
+    """
+
+    def __init__(
+        self,
+        referenced_images: Sequence[Dataset],
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        content_label: str,
+        content_description: Optional[str] = None,
+        graphic_annotations: Optional[Sequence[GraphicAnnotation]] = None,
+        graphic_layers: Optional[Sequence[GraphicLayer]] = None,
+        graphic_groups: Optional[Sequence[GraphicGroup]] = None,
+        concept_name_code: Union[Code, CodedConcept, None] = None,
```

Comment:
I would not append `_code`. We don't use this anywhere else either. The type hint should suffice

Reply:
Addressed in ea08248